src/main/java/com/microsoft/spring/data/gremlin/conversion/script/AbstractGremlinScriptLiteral.java

```diff
@@ -19,23 +19,25 @@
 
 public abstract class AbstractGremlinScriptLiteral {
 
-    protected static String generateEntityWithRequiredId(@NonNull Object id, GremlinEntityType type) {
+    protected static String generateEntityWithRequiredId(Object id, GremlinEntityType type) {
         Assert.isTrue(type == GremlinEntityType.EDGE || type == GremlinEntityType.VERTEX, "should be edge/vertex type");
 
         final String prefix = (type == GremlinEntityType.VERTEX) ? "V" : "E";
+        if (id != null) {
```

Comment:
the ID optional should not change the requiredId generation, should take care of id optional outside the caller.